tmt/checks/coredump.py

```diff
@@ -0,0 +1,350 @@
+import dataclasses
+import re
+from re import Pattern
+from typing import TYPE_CHECKING, Optional
+
+import tmt.log
+import tmt.steps.execute
+import tmt.steps.provision
+import tmt.utils
+from tmt.checks import Check, CheckEvent, CheckPlugin, _RawCheck, provides_check
+from tmt.result import CheckResult, ResultOutcome
+from tmt.steps.provision import GuestCapability
+from tmt.utils import Command, field
+
+if TYPE_CHECKING:
+    import tmt.base
+    from tmt.steps.execute import TestInvocation
+    from tmt.steps.provision import Guest
+
+DEFAULT_TMP_PATH = "/var/tmp/tmt"  # noqa: S108
+
+
+@dataclasses.dataclass
+class CoredumpCheck(Check):
+    """Configuration for the coredump check."""
+
+    # Patterns to ignore in crash reports
+    ignore_patterns: list[Pattern[str]] = field(
+        default_factory=list,
+        help="""
+             Optional list of regular expressions to ignore in crash reports.
+             If a crash report matches any of these patterns, it will be ignored
+             and not cause a failure. Any other crashes will still cause the test
+             to fail. If no patterns are specified, any crash will cause a failure.
+             """,
+        metavar="PATTERN",
+        normalize=tmt.utils.normalize_pattern_list,
+        exporter=lambda patterns: [pattern.pattern for pattern in patterns],
+        serialize=lambda patterns: [pattern.pattern for pattern in patterns],
+        unserialize=lambda serialized: [re.compile(pattern) for pattern in serialized],
+        )
+
+    # Internal flag to track if we have required capabilities
+    _has_capabilities: bool = True
+
+    def to_spec(self) -> _RawCheck:
+        """Convert to raw specification."""
+        spec = super().to_spec()
+
+        spec["ignore-patterns"] = [  # type: ignore[reportGeneralTypeIssues,typeddict-unknown-key,unused-ignore]
+            pattern.pattern for pattern in self.ignore_patterns
+            ]
+
+        return spec
+
+    def to_minimal_spec(self) -> _RawCheck:
+        """Convert to minimal raw specification."""
+        return self.to_spec()
+
+    def _configure_coredump(
+            self, guest: "Guest", logger: tmt.log.Logger, check_files_path: str
+            ) -> bool:
+        """Configure coredump storage."""
+        try:
+            # This should be default anyway
+            config = """[Coredump]
+Storage=external
+Compress=yes
+ProcessSizeMax=32G
+ExternalSizeMax=32G
+JournalSizeMax=767M
+MaxUse=0
+KeepFree=0
+"""
```

Comment:
I would make this a module-level string, with some nice name. It's pretty much a multiline constant, method indentation doesn't suit it.

Reply:
Addressed in c0c05df176604cf7fa3ec96d92f952e03b886726